### PR TITLE
testing: set pprof labels for goroutines that use the same code for different cases

### PR DESCRIPTION
**Pull request by cce**
This sets a few [profiler labels](https://rakyll.org/profiler-labels/) to identify a selection of goroutines used by algod. Since profiler data already lets you filter by function, we only need labels for cases where the same function is used by multiple goroutines for different cases. (E.g. low-prority worker pool vs high-priority worker pool; dedicated deserialization goroutines for different message types)

### CODE REVIEW

**Review by winder — DISMISSED** [review]
Neat. The docs on pprof.Do weren't really helpful for me, and I haven't spent a lot of time with profiling. It looks correct according to the blog post.

File: node/node.go [hunk 1]
```diff
@@ -381,8 +384,9 @@ func (node *AlgorandFullNode) startMonitoringRoutines() {
 	node.monitoringRoutinesWaitGroup.Add(2)
 	go node.txPoolGaugeThread(node.ctx.Done())
 	// Delete old participation keys
-	go node.oldKeyDeletionThread(node.ctx.Done())
-
+	pprof.Do(context.Background(), pprof.Labels("worker", "oldKeyDeletionThread"), func(_ context.Context) {
```
Comment (jannotti) [inline node/node.go:387]: Why doesn't the txPoolGaugeThread get labeled?

**Review by jannotti — COMMENTED** [review]
How did you decide what to label? `git grep "go.*()"` and dig around? Or stuff you knew was interesting?

File: node/node.go (hunk 1 above)
Comment (cce) [reply node/node.go:387]: It doesn't do much work ... I guess I should wrap it up for completeness though, it calls into the TransactionPool to ask for its size, so I might miss something if that is expensive.

Comment (jannotti) [reply node/node.go:387]: Is it possible for a thread/goroutine to label itself? It seems slightly better for the go routine itself to have a "label me" line as its first line of code, than to change call sites.

Comment (cce) [reply node/node.go:387]: I was thinking that too. the official pprof docs say to always use the `Do` wrapper but I think calling the [pprof.SetGoroutineLabels](https://pkg.go.dev/runtime/pprof#SetGoroutineLabels) function directly inside the goroutine itself is a little nicer-looking for us.

File: util/util.go [hunk 2]
```diff
@@ -72,3 +74,8 @@ func GetCurrentProcessTimes() (utime int64, stime int64, err error) {
 	}
 	return
 }
+
+// SetGoroutineLabels sets profiler labels for identifying goroutines using the pprof package.
+func SetGoroutineLabels(args ...string) {
```
Comment (algorandskiy) [inline util/util.go:79]: This file appears to be unix-only - see its build tags `//go:build !windows`. So Windows build failed.

```
Error: network\wsNetwork.go:1134:7: undefined: util.SetGoroutineLabels
```

**Review by algorandskiy — APPROVED** [review]

**Review by jasonpaulos — APPROVED** [review]

**Comment by algorandskiy** [comment]
I suggest not merging before p2p branch due to conflicts

**Pull request merged**